WordPress/Classes/Services/AccountSettingsService.swift

```diff
@@ -151,6 +151,13 @@ class AccountSettingsService {
         return notificationObserver.map(getSettings).startWith(getSettings())
     }
 
+    func primarySiteNameForSettings(settings: AccountSettings?) -> String {
```

Comment:
I'm not sure that's a good reason. I think public methods shouldn't be that coupled to their callers. if settings is nil, there's no point in calling `primarySiteNameForSettings` since the return will always be `nil`. That's what `map` is for. As for the return value, if you return a non-optional string it hides the fact that the method might fail: it will return a successful (empty) value even if it can't find a blog with the given ID. 

> why is mapViewModel relying on an optional model?

Because we might not always have the `settings` for an account. I think it was a bit of a shortcut, and I'm exploring [some alternatives](https://github.com/wordpress-mobile/WordPress-iOS/blob/db02ce6545cb2205f1e62d89f463a9669a77a011/WordPress/Classes/ViewRelated/Plans/PlanListViewController.swift#L73) in plans with an `enum` view model that has Loading/Ready/Error states.

Although in this particular case, the design called for showing the table with empty values while loading or after an error, so the mapping for all the states would look too similar.

Oh, I just remembered what the "shortcut" is: when `settings == nil`, the rows shouldn't be `EditableTextRow` but `TextRow` objects, as you shouldn't be able to edit them until we have some data.